### PR TITLE
DTFS2-5710 : Cover start date validation

### PR DESCRIPTION
**Pull request by abhi-markan**
## Bug
Upon facility issuance, if the facility cover start date was previously saved but not submitted an error message is thrown stating **cover start date cannot precede issue date** despite the fact that the cover start date is set to the issuance date.

This was due to cover start date selection was being ignored.

## Resolution
* `shouldCoverStartOnSubmission` is added to compound conditions.
* `coverStartDate` property in the return object returns issue date if `shouldCoverStartOnSubmission` is true otherwise user defined cover start date.
* Added Jest unit test.
* E2E test introducted.

### CODE REVIEW

File: gef-ui/server/controllers/unissued-facilities/validation.js [hunk 1]
```diff
@@ -220,7 +222,10 @@ const facilityValidation = async (body, query, params) => {
   }
 
   if (coverStartDateIsFullyComplete) {
-    coverStartDate = set(new Date(), { year: coverStartDateYear, month: coverStartDateMonth - 1, date: coverStartDateDay });
+    const value = shouldCoverStartOnSubmission
```
Comment (IgnatG) [inline gef-ui/server/controllers/unissued-facilities/validation.js:225]: can we use an `if/else` statement for this? the ternary operator gets harder to read if it's too complex

**Review by IgnatG — COMMENTED** [review]
It looks good. 
You need an E2E test though - or update an existing one to include this

File: gef-ui/server/controllers/unissued-facilities/validation.js (hunk 1 above)
Comment (abhi-markan) [reply gef-ui/server/controllers/unissued-facilities/validation.js:225]: Have updated.

File: gef-ui/server/controllers/unissued-facilities/validation.js [hunk 2]
```diff
@@ -220,7 +222,22 @@ const facilityValidation = async (body, query, params) => {
   }
 
   if (coverStartDateIsFullyComplete) {
-    coverStartDate = set(new Date(), { year: coverStartDateYear, month: coverStartDateMonth - 1, date: coverStartDateDay });
+    let value;
```
Comment (IgnatG) [inline gef-ui/server/controllers/unissued-facilities/validation.js:225]: you can assign a default value to `value`. and override it inside the `if` statement. in this scenario you wouldn't need the `else` part

**Comment by abhi-markan** [comment]
> It looks good. You need an E2E test though - or update an existing one to include this

That's all added.

File: gef-ui/server/controllers/unissued-facilities/index.test.js [hunk 3]
```diff
@@ -209,7 +211,7 @@ describe('postChangeUnissuedFacility()', () => {
         coverEndDate: format(tomorrow, 'MMMM d, yyyy'),
         coverStartDate: format(tomorrow, 'MMMM d, yyyy'),
         issueDate: format(now, 'MMMM d, yyyy'),
-        shouldCoverStartOnSubmission: null,
+        shouldCoverStartOnSubmission: false,
```
Comment (IgnatG) [inline gef-ui/server/controllers/unissued-facilities/index.test.js:214]: should this be a string? i.e. `false`
I've seen in the validation file that you are using something like `shouldCoverStartOnSubmission === 'false'`

Comment (IgnatG) [inline gef-ui/server/controllers/unissued-facilities/index.test.js:214]: should this be a string? i.e. `false`
I've seen in the validation file that you are using something like `shouldCoverStartOnSubmission === 'false'`

Comment (IgnatG) [reply gef-ui/server/controllers/unissued-facilities/index.test.js:214]: ah. never mind. that's the payload that we send to the server

Comment (IgnatG) [reply gef-ui/server/controllers/unissued-facilities/index.test.js:214]: ah. never mind. that's the payload that we send to the server

**Review by IgnatG — APPROVED** [review]

**Pull request merged**